File: codingTest/baekjoon/python/2484.py
import sys

rl = sys.stdin.readline


def showMeTheMoney(l):
    s = set(l)
    if len(s) == 1:
        return 50000 + (l[0] * 5000)
    ls = sorted(l)
    if len(s) == 2:
        if ls[1] == ls[2]:
            return 10000 + (ls[1] * 1000)
        else:
            return 2000 + (ls[0] * 500) + (ls[2] * 500)
    if len(s) == 3:
        for i in range(len(ls)):
            if ls[i] == ls[i + 1]:
                return 1000 + (ls[i] * 100)
    return ls[3] * 100


arr = []
for _ in range(int(rl())):
    arr.append(list(map(int, rl().split())))

result = []
for l in arr:
    result.append(showMeTheMoney(l))
print(result)
print(max(result))

File: codingTest/baekjoon/javascript/2484.js
function money(list) {
  list.sort((a, b) => a - b);
  var set = new Set(list);
  if (set.size == 1) return 50000 + list[0] * 5000;
  if (set.size == 2) return list[1] == list[2] ? 10000 + list[1] * 1000 : 2000 + list[1] * 500 + list[2] * 500;
  if (set.size == 3) for (let i in list) if (list[i] == list[+i + 1]) return 1000 + list[i] * 100;
  return list[3] * 100;
}
var rl = require("readline").createInterface({ input: process.stdin, output: process.stdout });
var input = [];
rl.on("line", line => {
  input.push(line);
}).on("close", () => {
  let result = [];
  for (let i = 1; i < input.length; i++) result.push(money(input[i].split(" ")));
  console.log(Math.max(...result));
});
